common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +461,100 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
     return true;
 }
 
+bool mon_wlan_hal_dwpal::channel_scan_trigger(int dwell_time_msec,
+                                              const std::vector<unsigned int> &channel_pool)
+{
+    LOG(DEBUG) << __func__ << " received on interface=" << m_radio_info.iface_name;
+
+    //build scan parameters
+    ScanParams channel_scan_params = {0};
+    sScanCfgParams org_fg, new_fg;   //foreground scan param
+    sScanCfgParamsBG org_bg, new_bg; //background scan param
+
+    // get original scan params
+    if (!dwpal_get_scan_params_fg(org_fg) || !dwpal_get_scan_params_bg(org_bg)) {
+        LOG(ERROR) << "Failed getting original scan parameters";
+        return false;
+    }
+
+    // prepare new scan params with changed dwell time
+    memcpy_s(&new_fg, sizeof(new_fg), &org_fg, sizeof(org_fg));
+    memcpy_s(&new_bg, sizeof(new_bg), &org_bg, sizeof(org_bg));
+    new_fg.passive = dwell_time_msec;
+    new_fg.active  = dwell_time_msec;
+    new_bg.passive = dwell_time_msec;
+    new_bg.active  = dwell_time_msec;
+
+    // set new scan params & get newly set values for validation
+    if (!dwpal_set_scan_params_fg(new_fg) || !dwpal_get_scan_params_fg(new_fg)) {
+        LOG(ERROR) << "Failed setting new values, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+    if (!dwpal_set_scan_params_bg(new_bg) || !dwpal_get_scan_params_bg(new_bg)) {
+        LOG(ERROR) << "Failed getting new scan parameters";
+        return false;
+    }
+
+    if ((new_fg.active != dwell_time_msec) || (new_fg.passive != dwell_time_msec) ||
+        (new_bg.active != dwell_time_msec) || (new_bg.passive != dwell_time_msec)) {
+        LOG(ERROR) << "Validation failed, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+
+    // get frequancies from channel pool and set in scan_params
+    if (!dwpal_get_freq(channel_pool, m_radio_info.channel, m_radio_info.iface_name, channel_scan_params)) {
+        LOG(ERROR) << "Failed getting frequencies, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+    LOG(DEBUG) << __func__ << " frequancies set in scan_params";
+    // must as single wifi won't allow scan on ap without this flag
+    channel_scan_params.ap_force = 1;
+
+    if (dwpal_driver_nl_scan_trigger(get_dwpal_nl_ctx(), (char *)m_radio_info.iface_name.c_str(),
+                                     &channel_scan_params) != DWPAL_SUCCESS) {
+        LOG(ERROR) << " scan trigger failed! Abort scan, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+
+    // timeout for nl confirmation on scan trigger
```

Comment:
Please elaborate a bit more in the comment - this needs explaining on why we do sleep.

Reply:
@itayx please  remove the sleep . we will add it later in validation stage with proper comment if needs to .